keel-api/src/main/kotlin/com/netflix/spinnaker/keel/api/artifacts/DeliveryArtifact.kt

```diff
@@ -32,6 +41,8 @@ abstract class DeliveryArtifact {
   abstract val reference: String // friendly reference to use within a delivery config
   abstract val deliveryConfigName: String? // the delivery config this artifact is a part of
   open val statuses: Set<ArtifactStatus> = emptySet()
+  open val branch: String? = null
```

Comment:
When the `BRANCH_AND_TIMESTAMP` sort method is used, is an artifact always associated with exactly one branch? Does that mean you need a separate artifact in the delivery config if you want to deploy a feature branch for testing?

(Not criticizing this approach, just asking for clarification. I don't actually know how deploying feature branches works currently).

Reply:
Yes, and it is always true that you need to define a new artifact if you want to deploy a new feature branch.

Reply:
That's right. Here's an example with release statuses when sorting by versions:
```yaml
artifacts:
- name: "@mycompany/myapp"
  reference: feature-branch
  type: npm
  statuses:
    - SNAPSHOT
- name: "@mycompany/myapp"
  reference: master
  type: npm
  statuses:
    - RELEASE
```
With branch and timestamp:
```yaml
artifacts:
- name: "@mycompany/myapp"
  reference: feature-branch
  type: npm
  branch: my-feature-branch
  sortBy: branch-and-timestamp
- name: "@mycompany/myapp"
  reference: master
  type: npm
  branch: master
  sortBy: branch-and-timestamp
```

Reply:
back to my question above - do we even need to specify the `sortBy?

Reply:
See response to previous comment.